dbms/src/Flash/Planner/PhysicalPlanNode.h

```diff
@@ -54,7 +58,11 @@ class PhysicalPlanNode
 
     virtual size_t childrenSize() const = 0;
 
-    virtual void transform(DAGPipeline & pipeline, Context & context, size_t max_streams);
+    virtual void buildBlockInputStream(DAGPipeline & pipeline, Context & context, size_t max_streams);
+
+    virtual void buildPipelineExec(PipelineExecGroupBuilder & /*group_builder*/, Context & /*context*/, size_t /*concurrency*/);
+
+    virtual void buildPipelines(const PipelinePtr & cur_pipeline, PipelineBuildState & state);
 
     virtual void finalize(const Names & parent_require) = 0;
     void finalize();
```

Comment:
Why 2 finalize here?

Reply:
The second `finalize` is
```
void PhysicalPlanNode::finalize()
{
    finalize(DB::toNames(schema));
}
```
just use for set the default argments.